cliquet/views/errors.py

```diff
@@ -67,9 +67,15 @@ def page_not_found(request):
                                    len(request.path)):]
         redirect = '%s%s' % (path, querystring)
         return HTTPTemporaryRedirect(redirect)
-    elif request.path.startswith('/' + request.registry.route_prefix):
+    elif (request.path.startswith('/' + request.registry.route_prefix + '/') or
+          not trailing_slash_redirection_enabled):
```

Comment:
nit: as a rule of thumb, never write a if on several lines. Please use intermediary variables :)